kolena/dataset/evaluation.py

```diff
@@ -112,14 +112,28 @@ def download_results(
     model: str,
 ) -> Tuple[pd.DataFrame, List[Tuple[EvalConfig, pd.DataFrame]]]:
     """
-    Fetch results given dataset name and model name.
+    Download results given dataset name and model name.
+
+    Concat dataset with results:
+
+    ```python
+    import kolena
+    import pandas as pd
+    from kolena.dataset import download_results
+
+    kolena.initialize(verbose=True)
+
+    df_dp, results = download_results("dataset name", "model name")
+    for eval_config, df_result in results:
+        df_combined = pd.concat([df_dp, df_result], axis=1)
```

Comment:
What will the behavior be here if we have a dataset with 100 datapoints, and a model that only has results for 10 datapoints?

Reply:
we will return empty results as placeholder for the missing dp results, in other words, dp and results will always align in row.

```
       city  height                                             locator  user_dp_id  width  score                                          softmax_bitmap
0  new york     405  https://fake-locator/3C1204PA1RIE/datapoints/5.png           5    505    0.0  https://fake-locator/3C1204PA1RIE/softmax_bitmap/0.png
1  waterloo     406  https://fake-locator/3C1204PA1RIE/datapoints/6.png           6    506    0.1  https://fake-locator/3C1204PA1RIE/softmax_bitmap/1.png
2  waterloo     407  https://fake-locator/3C1204PA1RIE/datapoints/7.png           7    507    0.2  https://fake-locator/3C1204PA1RIE/softmax_bitmap/2.png
3  waterloo     408  https://fake-locator/3C1204PA1RIE/datapoints/8.png           8    508    0.3  https://fake-locator/3C1204PA1RIE/softmax_bitmap/3.png
4  new york     409  https://fake-locator/3C1204PA1RIE/datapoints/9.png           9    509    0.4  https://fake-locator/3C1204PA1RIE/softmax_bitmap/4.png
5  new york     400  https://fake-locator/3C1204PA1RIE/datapoints/0.png           0    500    NaN                                                     NaN
6  waterloo     401  https://fake-locator/3C1204PA1RIE/datapoints/1.png           1    501    NaN                                                     NaN
7  new york     402  https://fake-locator/3C1204PA1RIE/datapoints/2.png           2    502    NaN                                                     NaN
8  waterloo     403  https://fake-locator/3C1204PA1RIE/datapoints/3.png           3    503    NaN                                                     NaN
9  waterloo     404  https://fake-locator/3C1204PA1RIE/datapoints/4.png           4    504    NaN                                                     NaN
```

Reply:
we expect that we will have `nan` as a placeholder for missing results, in other words, `df_dp` and `df_result` should always have the same shape and the concat logic will continue work. this behaviour is captured at `test__upload_results__missing_result`.

just noticed a bug for multiple config partial uploading, will fix it [in the server side first](https://linear.app/kolena/issue/KOL-4615/partial-results-for-multiple-configs-are-not-returning-properly) then update the integration at here